src/mnemonic/mod.rs

```diff
@@ -15,11 +15,13 @@ fn get_wordlist(language: Language) -> WordList {
     }
 }
 
-/// Converts a 12 word mnemonic to a entropy that can be used to
-/// generate a keypair
+/// Converts a 12/24 word mnemonic to entropy that can be used to
+/// generate a keypair. Checksum verification is disabled because the
+/// mobile wallet always has a '0000' for checksum while valid BIP39
+/// mnemonics will have a valid checksum
 pub fn mnemonic_to_entropy(words: Vec<String>) -> Result<[u8; 32]> {
-    if words.len() != 12 {
-        bail!("Invalid number of seed words");
+    if words.len() != 12 && words.len() != 24 {
```

Comment:
This (and how I handle entropy_bytes below) are kind of a hack hard coded for only two options, not the full range of BIP-39 mnemonics. I considered being more flexible here and below, but thought quicker to PR was a better choice for my first attempt at Rust.